### ISSUE TITLE
Rule Change: `padding-line-between-statements` be able to set a new statement_type `assignment`

### ISSUE THREAD

**Issue by AdaLimana** [eslint/eslint#17445]
### What rule do you want to change?

padding-line-between-statements)

### What change to do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
a = 1;
b = foo();
bar();
```

### What does the rule currently do for this code?

a = 1; //select this statement
b = foo(); //select this statement
bar();

### What will the rule do after it's changed?

Allow to select assignments statements to set up blank line rules.

For example:
```
 options: [
     { blankLine: "always", prev: "assignment", next: "*" },
     { blankLine: "always", prev: "*", next: "assignment" },
     { blankLine: "never", prev: "assignment", next: "assignment" }
 ]
```
### Participation

- [X] I am willing to submit a pull request to implement this change.

**Comment by Rec0iL99** [comment]
Hi @AdaLimana, thanks for the issue. You can catch assignment statements using the `expression` type.

[playground](https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgcGFkZGluZy1saW5lLWJldHdlZW4tc3RhdGVtZW50czogW1xuICAgIFwiZXJyb3JcIixcbiAgICB7IGJsYW5rTGluZTogXCJhbHdheXNcIiwgcHJldjogXCJleHByZXNzaW9uXCIsIG5leHQ6IFwiZXhwcmVzc2lvblwiIH1cbl0qL1xuXG5hID0gMTtcbmIgPSBmb28oKTtcbmJhcigpOyIsIm9wdGlvbnMiOnsiZW52Ijp7ImVzNiI6dHJ1ZX0sInJ1bGVzIjp7fSwicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnt9LCJlY21hVmVyc2lvbiI6ImxhdGVzdCIsInNvdXJjZVR5cGUiOiJtb2R1bGUifX19)

Plus, we currently aren't adding any new options to stylistic rules due to the [policy change](https://github.com/eslint/eslint/tree/main#stylistic-rule-updates). You can always copy the existing core rule into a custom rule and add new options to it.

**Comment by AdaLimana** [comment]
Hi @Rec0iL99 , thanks for the advice. But there is a problem when using `expression` to catch assignment statements because it catches expressions that aren't assignment, like function call. And I thought that since assignments are another type of statement, it would be worth having this option."

[playground](https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgcGFkZGluZy1saW5lLWJldHdlZW4tc3RhdGVtZW50czogW1xuICAgIFwiZXJyb3JcIixcbiAgICB7IGJsYW5rTGluZTogXCJhbHdheXNcIiwgcHJldjogXCJleHByZXNzaW9uXCIsIG5leHQ6IFwiKlwiIH0sXG4gICAgeyBibGFua0xpbmU6IFwiYWx3YXlzXCIsIHByZXY6IFwiKlwiLCBuZXh0OiBcImV4cHJlc3Npb25cIiB9LFxuICAgIHsgYmxhbmtMaW5lOiBcIm5ldmVyXCIsIHByZXY6IFwiZXhwcmVzc2lvblwiLCBuZXh0OiBcImV4cHJlc3Npb25cIiB9XG5dKi9cblxuYSA9IDE7XG5iID0gZm9vKCk7XG5cbmJhcigpOyIsIm9wdGlvbnMiOnsiZW52Ijp7ImVzNiI6dHJ1ZX0sInJ1bGVzIjp7fSwicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnt9LCJlY21hVmVyc2lvbiI6ImxhdGVzdCIsInNvdXJjZVR5cGUiOiJtb2R1bGUifX19).

However, I understand if it's not possible to add it due to policy.
